docs/ceph-csi-upgrade.md

```diff
@@ -291,6 +299,74 @@ service/csi-metrics-rbdplugin configured
 
 we have successfully upgraded RBD csi from v3.6 to v3.7
 
+### Upgrading NFS
+
+Upgrading nfs csi includes upgrade of nfs driver and as well as
+kubernetes sidecar containers and also the permissions required for the
+kubernetes sidecar containers, lets upgrade the things one by one
+
+#### 5. Upgrade NFS Provisioner resources
+
+Upgrade provisioner resources include updating the provisioner RBAC and
+Provisioner deployment
+
+##### 5.1 Update the NFS Provisioner RBAC
+
+```bash
+$ kubectl apply -f deploy/nfs/kubernetes/csi-provisioner-rbac.yaml
+serviceaccount/nfs-csi-provisioner configured
+clusterrole.rbac.authorization.k8s.io/nfs-external-provisioner-runner configured
+clusterrolebinding.rbac.authorization.k8s.io/nfs-csi-provisioner-role configured
+role.rbac.authorization.k8s.io/nfs-external-provisioner-cfg configured
+rolebinding.rbac.authorization.k8s.io/nfs-csi-provisioner-role-cfg configured
+```
+
+##### 5.2 Update the NFS Provisioner deployment
+
+```bash
+$ kubectl apply -f deploy/nfs/kubernetes/csi-nfsplugin-provisioner.yaml
+service/csi-nfsplugin-provisioner configured
+deployment.apps/csi-nfsplugin-provisioner configured
+```
+
+wait for the deployment to complete
+
+```bash
+$ kubectl get deployment
+NAME                           READY   UP-TO-DATE   AVAILABLE   AGE
+csi-nfsplugin-provisioner      5/5     1            5           104m
+```
+
+deployment UP-TO-DATE value must be same as READY
+
+#### 6. Upgrade NFS Nodeplugin resources
+
+Upgrading nodeplugin resources include updating the nodeplugin RBAC and
+nodeplugin daemonset
+
+##### 6.1 Update the NFS Nodeplugin RBAC
+
+```bash
+$ kubectl apply -f deploy/nfs/kubernetes/csi-nodeplugin-rbac.yaml
+serviceaccount/nfs-csi-nodeplugin configured
+```
+
+##### 6.2 Update the NFS Nodeplugin daemonset
+
+```bash
+$ kubectl apply -f deploy/nfs/kubernetes/csi-nfsplugin.yaml
+daemonset.apps/csi-nfsplugin configured
+service/csi-metrics-nfsplugin configured
+```
+
+##### 6.3 Delete the old NFS Nodeplugin daemonset
+
+```bash
+$ kubectl delete daemonsets.apps csi-nfs-node
```

Comment:
add `-ignore-not-found` flag so that we dont need to worry about it in the next release?

Reply:
> add `-ignore-not-found` flag so that we dont need to worry about it in the next release?

Let's remove it for upgrade from 3.7 to 3.8 point of time.

Reply:
https://github.com/ceph/ceph-csi/issues/3324